### PR TITLE
[Workspace] Make non-editable checkouts immutable

### PR DESCRIPTION
**Pull request by aciidb0mb3r**
-- <rdar://problem/31286403> Unedited Swift package sources should be read-only

### CODE REVIEW

**Comment by aciidb0mb3r** [comment]
@swift-ci please smoke test

**Comment by aciidb0mb3r** [comment]
@swift-ci please smoke test

**Comment by aciidb0mb3r** [comment]
@swift-ci please smoke test

**Comment by aciidb0mb3r** [comment]
@swift-ci please smoke test

**Comment by aciidb0mb3r** [comment]
@swift-ci please smoke test

File: Sources/Workspace/Workspace.swift [hunk 1]
```diff
@@ -347,10 +347,16 @@ public class Workspace {
...
     public func reset(with diagnostics: DiagnosticsEngine) {
-        guard diagnostics.wrap({ try managedDependencies.reset() }) else {
-            return
-        }
-        
+        let removed = diagnostics.wrap({ 
+            // Make all checkouts mutable.
+            try fileSystem.set(attribute: .mutable, path: checkoutsPath, recursive: true)
+
+            // Reset manaked dependencies.
```
Comment (heckj) [inline Sources/Workspace/Workspace.swift]: typo in comment - "manaked" probably should be "managed"

**Comment by aciidb0mb3r** [comment]
@heckj We want to do this so it is not possible for users to edit the checkouts by hand. We have a dedicated feature for editing. It is still possible for users to remove the immutable flag and edit the checkout anyway but by then it would be clear that they are going against the policies.

This is also related to [SR-1439](https://bugs.swift.org/browse/SR-1439).

**Pull request merged**